rust/ql/lib/codeql/rust/internal/TypeInference.qll

```diff
@@ -690,7 +690,7 @@ private Type inferCallExprBaseType(AstNode n, TypePath path) {
   |
     if apos.isSelf()
     then
-      exists(Type receiverType | receiverType = CallExprBaseMatchingInput::inferReceiverType(n) |
```

Comment:
I don't remember why it was done like this originally, but if it doesn't impact performance, this is fine, and we can now remove the two `inferReceiverType` predicates.

Reply:
Yes, I forgot that. Done now :)